shared/src/api/extension/flatExtentionAPI.ts

```diff
@@ -53,10 +79,24 @@ export const initNewExtensionAPI = (mainAPI: Remote<MainThreadAPI>): InitResult
         return configuration
     }
 
+    // Workspace
+    const workspace: PartialWorkspaceNamespace = {
+        get roots() {
+            return state.roots
+        },
+        get versionContext() {
+            return versionContextChanges.value
+        },
+        onDidChangeRoots: rootChanges,
+        rootChanges,
```

Comment:
```suggestion
        onDidChangeRoots: rootChanges.asObservable(),
        rootChanges: rootChanges.asObservable(),
```

Apparently this was unfortunately not done before, but these should be exposed as plain observables, like `versionContextChanges`, to avoid exposing Subject methods like `.next()` to extensions.